tests/Fixer/Import/FullyQualifiedStrictTypesFixerTest.php

```diff
@@ -1208,6 +1208,17 @@ class SomeClass {}',
 class SomeClass {}',
         ];
 
+        yield 'Test PHPDoc union' => [
```

Comment:
What's the purpose of this test case? If we want to shorten/import symbols from complex types in the future, then adding such a test is not a good idea because we shouldn't change it later? 🤔

Please help me understand what kind of contract it is.

Reply:
see https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7649#discussion_r1439508513 answer

we should not fix the 1st type in union in general, as such is leading to very inconsistent types like:
```diff
 <?php

-/* @var \A|\B */
+/* @var A|\B */
```

Reply:
I have improved the test to stress the current behaviour.